src/WaipuData.cpp

```diff
@@ -1038,15 +1038,26 @@ PVR_ERROR WaipuData::IsEPGTagRecordable(const kodi::addon::PVREPGTag& tag, bool&
 
 PVR_ERROR WaipuData::IsEPGTagPlayable(const kodi::addon::PVREPGTag& tag, bool& isPlayable)
 {
+  isPlayable = false;
   for (const auto& channel : m_channels)
   {
     if (channel.iUniqueId != tag.GetUniqueChannelId())
       continue;
     isPlayable = channel.tvfuse;
-    return PVR_ERROR_NO_ERROR;
+    if (isPlayable) {
+      return PVR_ERROR_NO_ERROR;
+    }
   }
 
-  isPlayable = false;
+  auto current_time = time(NULL);
+  if (current_time < tag.GetEndTime())
+  {
+    isPlayable = true;
+  }
+  if (tag.GetStartTime() > current_time)
+  {
+    isPlayable = false;
+  }
```

Comment:
```suggestion
  if (current_time > tag.GetStartTime() && current_time < tag.GetEndTime())
  {
    // tag is current program: allow replay  
    isPlayable = true;
    return PVR_ERROR_NO_ERROR;
  }
```

Can we simplify it sth like that? (Untested) 

Is replay available for every channel?

Reply:
Guess we could but we don't need the return there as there's another return down below.

I don't know. I tested a few of the major channels and it worked. I think for some shows replay might be disabled but it would just silently fail then so no issue.

I've enabled edits by maintainers so you can edit the code if you want.